[CI] org.elasticsearch.gradle.internal.ElasticsearchJavaModulePathPluginFuncTest

### CI Link

https://gradle-enterprise.elastic.co/s/xunq6rymvvy34/tests/:build-tools-internal:integTest/org.elasticsearch.gradle.internal.ElasticsearchJavaModulePathPluginFuncTest/included%20build%20with%20non%20module%20dependencies

### Repro line

`./gradlew :build-tools-internal:integTest`

### Does it reproduce?

Didn't try

### Applicable branches

main

### Failure history

https://gradle-enterprise.elastic.co/scans/tests?search.timeZoneId=Europe/Berlin&tests.container=org.elasticsearch.gradle.internal.ElasticsearchJavaModulePathPluginFuncTest&tests.test=included%20build%20with%20non%20module%20dependencies#

### Failure excerpt

```
Output:	
Configuration cache is an incubating feature.	
Calculating task graph as no configuration cache is available for tasks: compileJava	
	
FAILURE: Build failed with an exception.	
	
* Where:	
Settings file 'C:\Users\jenkins\AppData\Local\Temp\junit16491424318159097053\settings.gradle' line: 2	
	
* What went wrong:	
Could not compile settings file 'C:\Users\jenkins\AppData\Local\Temp\junit16491424318159097053\settings.gradle'.	
> startup failed:	
  settings file 'C:\Users\jenkins\AppData\Local\Temp\junit16491424318159097053\settings.gradle': 2: Unexpected character: '\'' @ line 2, column 22.	
             includeBuild 'C:\Users\jenkins\AppData\Local\Temp\junit8721228232480723469'	
                          ^	
  	
  1 error
```

--- a/build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/ElasticsearchJavaModulePathPluginFuncTest.groovy
+++ b/build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/ElasticsearchJavaModulePathPluginFuncTest.groovy
@@ -204,7 +204,7 @@ class ElasticsearchJavaModulePathPluginFuncTest extends AbstractJavaGradleFuncTe
     def "included build with module dependencies"() {
         given:
         file(rootBuild.root, 'settings.gradle') << """
-        includeBuild '$projectDir'
+        includeBuild '${projectDir.path.replace('\\', '\\\\')}'
         """
 
         file(rootBuild.root, 'build.gradle') << """

--- a/build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/ElasticsearchJavaModulePathPluginFuncTest.groovy
+++ b/build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/ElasticsearchJavaModulePathPluginFuncTest.groovy
@@ -172,7 +172,7 @@ class ElasticsearchJavaModulePathPluginFuncTest extends AbstractJavaGradleFuncTe
     def "included build with non module dependencies"() {
         given:
         file(rootBuild.root, 'settings.gradle') << """
-        includeBuild '$projectDir'
+        includeBuild '${projectDir.path.replace('\\', '\\\\')}'
         """
 
         file(rootBuild.root, 'build.gradle') << """

